src/DependencyInjection/Providers/AppleProviderConfigurator.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace KnpU\OAuth2ClientBundle\DependencyInjection\Providers;
+
+use Symfony\Component\Config\Definition\Builder\NodeBuilder;
+
+class AppleProviderConfigurator implements ProviderConfiguratorInterface
+{
+    public function buildConfiguration(NodeBuilder $node)
+    {
+        $node
+            ->scalarNode('team_id')
+                ->cannotBeEmpty()
+            ->end()
+            ->scalarNode('key_file_id')
+                ->cannotBeEmpty()
+            ->end()
+            ->scalarNode('key_file_path')
+                ->cannotBeEmpty()
+            ->end()
+        ;
+    }
+
+    public function getProviderClass(array $configuration)
+    {
+        return 'League\OAuth2\Client\Provider\Apple';
+    }
+
+    public function getClientClass(array $config)
+    {
+        return 'KnpU\OAuth2ClientBundle\Client\Provider\AppleClient';
+    }
+
+    public function getProviderOptions(array $configuration)
+    {
+        return [
+            'clientId' => $configuration['client_id'],
```

Comment:
Ah, wait, `client_secret` is missing here :)

Reply:
apple does not have a client_secret, they use the key_file instead

Reply:
In this case, we should remove `client_secret` from the docs I think